src/useIsDraggingWithin/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * WordPress specific abstraction layer atop React.
+ *
+ * @see    https://developer.wordpress.org/block-editor/reference-guides/packages/packages-element/
+ * @ignore
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Function to be called when component is mounted.
+ *
+ * @ignore
+ */
+import useDidMount from '../useDidMount';
+
+/**
+ * This hook determines whether an element is being dragged within another (ref) target element.
+ *
+ * @function
+ * @since      1.13.0
+ * @param  	   {Object}     elementRef    A mutable ref element object whose `.current` property is initialized to the passed argument.
+ * @return     {boolean}				  Is dragging within the target element.
+ * @example
+ *
+ * const listItemRef = useRef( null );
+ * const isDraggingWithin = useIsDraggingWithin( listItemRef );
+ */
+function useIsDraggingWithin( elementRef ) {
+	const [ isDraggingWithin, setIsDraggingWithin ] = useState( false );
+
+	useDidMount( () => {
+		const { ownerDocument } = elementRef.current;
+
+		// Check the first time when the dragging starts.
+		const handleDragStart = ( event ) => {
+			handleDragEnter( event );
+		};
+
+		// Set to false whenever the user cancel the drag event by either releasing the mouse or press Escape.
+		const handleDragEnd = () => {
+			setIsDraggingWithin( false );
+		};
+
+		// Check if the current target is inside the item element.
+		const handleDragEnter = ( event ) => {
+			if ( elementRef.current.contains( event.target ) ) {
```

Comment:
Just a tiny detail: since we're assigning `ownerDocument` on line 32, we would be able to reuse it here as well